packages/react/src/PlanDefinitionBuilder.tsx

```diff
@@ -0,0 +1,277 @@
+        </FormSection>
+        <ActionArrayBuilder actions={value.action || []} onChange={(x) => changeProperty('action', x)} />
+        <Button type="submit" size="large">
+          Save
+        </Button>
+      </Form>
+    </div>
+  );
+}
+
+interface ActionArrayBuilderProps {
+  actions: PlanDefinitionAction[];
+  onChange: (actions: PlanDefinitionAction[]) => void;
+}
+
+function ActionArrayBuilder(props: ActionArrayBuilderProps): JSX.Element {
+  const actionsRef = useRef<PlanDefinitionAction[]>();
+  actionsRef.current = props.actions;
+
+  function changeAction(changedAction: PlanDefinitionAction): void {
+    props.onChange(
+      (actionsRef.current as PlanDefinition[]).map((i) => (i.id === changedAction.id ? changedAction : i))
+    );
+  }
+
+  function addAction(addedAction: PlanDefinitionAction): void {
+    props.onChange([...(actionsRef.current as PlanDefinition[]), addedAction]);
+  }
+
+  function removeAction(removedAction: PlanDefinitionAction): void {
+    props.onChange((actionsRef.current as PlanDefinition[]).filter((i) => i !== removedAction));
+  }
+
+  return (
+    <div className="section">
+      {props.actions.map((i) => (
+        <div key={i.id}>
+          <ActionBuilder action={i} onChange={changeAction} onRemove={() => removeAction(i)} />
+        </div>
+      ))}
+      <div className="bottom-actions">
+        <a
+          href="#"
+          onClick={(e) => {
+            e.preventDefault();
+            addAction({ id: generateId() });
+          }}
+        >
+          Add action
+        </a>
+      </div>
+    </div>
+  );
+}
+
+interface ActionBuilderProps {
+  action: PlanDefinitionAction;
+  onChange: (action: PlanDefinitionAction) => void;
+  onRemove: () => void;
+}
+
+function ActionBuilder(props: ActionBuilderProps): JSX.Element {
+  const { action } = props;
+  const [actionType, setActionType] = useState<string | undefined>(getInitialActionType(action));
+
+  const actionRef = useRef<PlanDefinitionAction>();
+  actionRef.current = props.action;
+
+  function changeProperty(property: string, value: any): void {
+    props.onChange({
+      ...actionRef.current,
+      [property]: value,
+    } as PlanDefinitionAction);
+  }
+
+  return (
+    <div className="section">
+      <FormSection
+        title="Action Title"
+        description="The name of the action, an operational task to be completed."
+        htmlFor={`actionTitle-${action.id}`}
+      >
+        <Input
+          name={`actionTitle-${action.id}`}
+          defaultValue={action.title}
+          onChange={(newValue) => changeProperty('title', newValue)}
+        />
+      </FormSection>
+      <FormSection
+        title="Action Type"
+        description="The type of the action to be performed."
+        htmlFor={`actionType-${action.id}`}
+      >
+        <Select name={`actionType-${action.id}`} defaultValue={actionType} onChange={setActionType}>
```

Comment:
Can we make this text for all these options more descriptive? From the current text, it's hard to tell what kind of action is required.

Reply:
I agree that the text is confusing, but I think we should review holistically once we have more clarity on the task types.  I'm not 100% sure what "Documentation" / "Request for Documentation" will result in.  I also want to make sure the text is consistent in "noun" vs "verb noun", etc.

I like the idea of adding more details in context... We probably need something like the Tailwind "[Branded with supported text](https://tailwindui.com/components/application-ui/forms/select-menus#component-d47b46d3ffea1f6b159f51be4bcae2ff)" component:

<img width="933" alt="image" src="https://user-images.githubusercontent.com/749094/176504547-8e73371b-3955-4d71-ac92-964fb9908ee1.png">